starport/pkg/cosmosaccount/cosmosaccount.go

```diff
@@ -0,0 +1,216 @@
+package cosmosaccount
+
+import (
+	"errors"
+	"os"
+
+	dkeyring "github.com/99designs/keyring"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/go-bip39"
+)
+
+// KeyringServiceName used for the name of keyring in OS backend.
+const KeyringServiceName = "starport"
+
+// KeyringHome used to store account related data.
+var KeyringHome = os.ExpandEnv("$HOME/.starport/accounts")
+
+var (
+	ErrAccountExists       = errors.New("account already exists")
+	ErrAccountDoesNotExist = errors.New("account does not exist")
+)
+
+const (
+	accountPrefixCosmos = "cosmos"
+	pubKeyPrefix        = "pub"
+)
+
+// Registry for accounts.
+type Registry struct {
+	kr keyring.Keyring
+}
+
+// New creates a new registry to manage accounts.
+func New(backend string) (Registry, error) {
+	kr, err := keyring.New(KeyringServiceName, backend, KeyringHome, os.Stdin)
+	if err != nil {
+		return Registry{}, err
+	}
+
+	r := Registry{
+		kr: kr,
+	}
+
+	return r, nil
+}
+
+// Account represents an CosmosSDK account.
+type Account struct {
+	// Name of the account.
+	Name string
+
+	// Info holds additional info about the account.
+	Info keyring.Info
+}
+
+// Address returns an address for acocunt for given prefix.
+func (a Account) Address(accPrefix string) string {
+	if accPrefix == "" {
+		accPrefix = accountPrefixCosmos
+	}
+
+	conf := types.GetConfig()
+	conf.SetBech32PrefixForAccount(accPrefix, pubKeyPrefix)
+
+	ko, err := keyring.Bech32KeyOutput(a.Info)
+	if err != nil {
+		panic(err)
+	}
+	return ko.Address
+}
+
+// PubKey returns a public key for given account prefix.
+func (a Account) PubKey(accPrefix string) string {
+	if accPrefix == "" {
+		accPrefix = accountPrefixCosmos
+	}
+
+	conf := types.GetConfig()
+	conf.SetBech32PrefixForAccount(accPrefix, pubKeyPrefix)
+
+	o, err := keyring.Bech32KeyOutput(a.Info)
+	if err != nil {
+		panic(err)
+	}
+	return o.PubKey
+}
+
+// Create creates a new account with name.
+func (r Registry) Create(name string) (acc Account, mnemonic string, err error) {
+	acc, err = r.GetByName(name)
+	if err == nil {
+		return Account{}, "", ErrAccountExists
+	}
+	if !errors.Is(err, ErrAccountDoesNotExist) {
+		return Account{}, "", err
+	}
+
+	entropySeed, err := bip39.NewEntropy(256)
```

Comment:
Do we always use 24-word mnemonics? Maybe we can add a flag to specify the number of words or the bitesize of the entropy.

Reply:
24 words safer I think, we don't necessarily need to let people choose a weaker mnemonic.